geopmdpy/geopmdpy/session.py

```diff
@@ -97,12 +111,15 @@ def run_read(self, requests, duration, period, pid, out_stream):
         signal_handles = []
         for name, dom, dom_idx in requests:
             signal_handles.append(pio.push_signal(name, dom, dom_idx))
-
         for sample_idx in loop.TimedLoop(period, num_period):
             pio.read_batch()
-            signals = [pio.sample(handle) for handle in signal_handles]
-            line = self.format_signals(signals, requests.get_formats())
-            out_stream.write(line)
+            if out_stream is not None:
+                signals = [pio.sample(handle) for handle in signal_handles]
+                line = self.format_signals(signals, requests.get_formats())
+                out_stream.write(line)
```

Comment:
Should we be flushing here? Example where it might matter:

`geopmsession -t 123 -p 1 <<< 'TIME board 0' | tee geopm-trace.csv` appears to do nothing for a while until the write buffer fills up enough to flush automatically

Reply:
Also relates to https://github.com/geopm/geopm/issues/3543#issuecomment-2276474324, where we get no report if a run is interrupted. For traces, the lack of signal handling means we get truncated output on non-interactive runs (which typically do not default to line-buffered)